src/Sylius/Component/Translation/Factory/TranslatableFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Translation\Factory;
+
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Factory\Factory;
+use Sylius\Component\Translation\Model\TranslatableInterface;
+use Sylius\Component\Translation\Provider\LocaleProviderInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+class TranslatableFactory extends Factory implements TranslatableFactoryInterface
```

Comment:
Anyway, if we rewrite this class as decorator, it is possible to make some vendor class translatable even without changing its factory code.